pennylane/qaoa/mixers.py

```diff
@@ -112,3 +114,85 @@ def xy_mixer(graph):
         obs.append(qml.PauliY(node1) @ qml.PauliY(node2))
 
     return qml.Hamiltonian(coeffs, obs)
+
+
+def bit_flip_mixer(graph, n):
+    r"""Creates a bit-flip mixer Hamiltonian.
+
+    This mixer is defined as:
+
+    .. math:: H_M \ = \ \displaystyle\sum_{v \in V(G)} \frac{1}{2^d(v)} X_{v}
+              \displaystyle\prod_{w \in \text{nbhd}(v)} (\mathbb{I} \ + \ (-1)^n Z_w)
```

Comment:
Minor thing: how about `N(v)` instead of `nbhd(v)`? This is what is used here: https://en.wikipedia.org/wiki/Neighbourhood_(graph_theory)